tests/issue0079.py

```diff
@@ -0,0 +1,39 @@
+from __future__ import absolute_import, print_function
+import unittest, time
+try:
+    from StringIO import StringIO
+except ImportError:
+    from io import StringIO
+
+class runner(object):
+    def runTest(self):
+        from pcpp import Preprocessor, OutputDirective, Action
+        import os, sys, re
+
+        p = Preprocessor()
+        p.parse(self.input)
+        oh = StringIO()
+        p.write(oh)
+        print("ok")
+        if oh.getvalue() != self.output:
+            print("Should be:\n" + self.output + "EOF\n", file = sys.stderr)
+            print("\nWas:\n" + oh.getvalue()+"EOF\n", file = sys.stderr)
+        self.assertEqual(p.return_code, 0)
+        self.assertEqual(oh.getvalue(), self.output)
+
+class pp_number_pasting(unittest.TestCase, runner):
+    input = r"""#define CAT_(A,B)A##B
```

Comment:
I'll switch to string with embedded `\n` newlines.
Guess that needs to be tested on Windows too :smiling_face_with_tear:

Reply:
Sorry, you misunderstood. I meant the `r` can go. Can be a multi-line string without any problem. But I can see why it may have been used as a precaution.

Reply:
I see now, thanks (I'm forgetting Python details and conflating with C++ that needs `r` and delimiter for raw strings)